remote_engine_client/src/cached_router.rs

```diff
@@ -19,12 +19,18 @@ pub struct CachedRouter {
 
     /// Cache mapping table to channel of its endpoint
     // TODO: we should add gc for the cache
-    cache: RwLock<HashMap<TableIdentifier, Channel>>,
+    cache: RwLock<HashMap<TableIdentifier, RouteContext>>,
 
     /// Channel pool
     channel_pool: ChannelPool,
 }
 
+#[derive(Clone)]
+pub struct RouteContext {
+    pub channel: TonicChannel,
```

Comment:
Does the `TonicChannel` contain any identifier for the endpoint?

Reply:
Not.